functions.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Beans Child
+ *
+ * @package Beans\StarterChildTheme
+ */
+
+namespace Beans\StarterChildTheme;
+
+// Include Beans. Do not remove the line below.
+require_once get_template_directory() . '/lib/init.php';
+
+/**
+ * Enqueue LESS and CSS style to the UIkit compiler.
+ *
+ * IMPORTANT: The function below enqueue both the style.less and style.css to the UIkit compiler. Remove one of the
+ * two enqueuer according to your needs
+ *
+ * To make sure the compiler re-compile your LESS or CSS on the fly (on page reload), make sure to enable development
+ * mode via the Admin->Appearance->Settings option.
+ */
+add_action( 'beans_uikit_enqueue_scripts', function() {
```

Comment:
I'm not a fan of using closures as the hooked callback.  Why? 

Our code provides an example of _how to build_ code.  And in some cases a closure is fine.  But in most others it is not because it cannot be unhooked.

Here, one could argue that this callback will never be unhooked.  But in many other cases, you may build a plugin that needs to unhook another part of your theme.  If you use a closure, that is no longer possible, well unless we use `beans_add_action`.

## Choices:

1. Use the standard function as the callback. 

```
add_action( 'beans_uikit_enqueue_scripts', __NAMESPACE__ . '\beans_child_enqueue_uikit' );
/**
 * Enqueue LESS and CSS style to the UIkit compiler.
 *
 * IMPORTANT: The function below enqueue both the style.less and style.css to the UIkit compiler. Remove one of the
 * two enqueuer according to your needs
 *
 * To make sure the compiler re-compile your LESS or CSS on the fly (on page reload), make sure to enable development
 * mode via the Admin->Appearance->Settings option.
 */
function beans_child_enqueue_uikit() {
```

2. Use `beans_add_action()` with the closure, thereby, giving the ability for a plugin to modify or remove the closure.

Reply:
The thinking behind is that Child Theme are not meant to be extendable to a certain extent, but I am definitely in a agreement that it can be the case sometimes. I think using `beans_add_action()` is a good showcase of how closure can safely be used with Beans.